pkg/serviceaccounttoken/secret_test.go

```diff
@@ -123,6 +264,7 @@ func TestEnsureSecretForServiceAccount(t *testing.T) {
 			assert.Equal(t, tt.wantSecret.Name, got.Name)
 			gotSA, _ := k8sClient.CoreV1().ServiceAccounts("default").Get(context.Background(), "test", metav1.GetOptions{})
 			assert.Equal(t, tt.wantSA.Secrets, gotSA.Secrets)
+			assert.Equal(t, tt.sa, gotSA)
```

Comment:
Wouldn't this comparison subsume the one above that compares just the lists of secret refs?

Reply:
No, because we're not comparing `tt.wantSA`, we're comparing the original, `tt.sa`, which needs to be updated as a side effect. Before this change, the original object was continuing to be referenced but was out of date after this function was run.